care/facility/api/serializers/asset.py

```diff
@@ -53,7 +51,19 @@ def validate(self, attrs):
 
         return super().validate(attrs)
 
+    def create(self, validated_data):
+        camera_details = validated_data["meta"]["camera"]["url"] and validated_data["meta"]["camera"]["preset"]
+
+        if validated_data["asset_type"] == Asset.AssetType.CAMERA.value and not camera_details:
+            raise ValidationError({"Camera Details": "Cannot be blank when asset type is Camera"})
+        return super().create(validated_data)
+
     def update(self, instance, validated_data):
+        camera_details = validated_data["meta"]["camera"]["url"] and validated_data["meta"]["camera"]["preset"]
```

Comment:
since the `camera` attribute is specific to the asset type camera, there can be cases where the key `camera` is not 
 present, this will produce a key error, move the check inside the if block or use the built-in get method to get the value or add an exception handling block.

Reply:
Since we are repeating the same validation in create and update, it also makes sense to make it into a function and abstracting it out.